vars/publishIntegTestResults.groovy

```diff
@@ -88,8 +97,159 @@ void call(Map args = [:]) {
     indexFailedTestData(indexName, "test-records.json")
 }
 
+def processFailedTests(failedTests, componentName, componentRepo, componentRepoUrl, version, integTestBuildNumber, 
+                       integTestBuildUrl, distributionBuildNumber, distributionBuildUrl, buildStartTime, 
+                       rc, rcNumber, platform, architecture, distribution, componentCategory, securityType, testFailuresindexName) {
+
+    def finalFailedTestsJsonDoc = ""
+    switch (true) {
+        case failedTests.isEmpty():
+            break
+        case failedTests.contains("Test Result Not Available"):
+            def testResultJsonContent = generateFailedTestJson(componentName, componentRepo, componentRepoUrl, version, integTestBuildNumber, 
+                integTestBuildUrl, distributionBuildNumber, distributionBuildUrl, buildStartTime, rc, rcNumber, 
+                platform, architecture, distribution, componentCategory, securityType, "Not Available", "Not Available")
+            finalFailedTestsJsonDoc += "{\"index\": {\"_index\": \"${testFailuresindexName}\"}}\n${testResultJsonContent}\n"
+            break
+        case failedTests.contains("No Failed Test"):
+            break
+        default:
+            failedTests.collect { failedTest ->
+                def match = failedTest.split("#")
+                if (match) {
+                    def testResultJsonContent = generateFailedTestJson(componentName, componentRepo, componentRepoUrl, version, integTestBuildNumber, 
+                        integTestBuildUrl, distributionBuildNumber, distributionBuildUrl, buildStartTime, rc, rcNumber, 
+                        platform, architecture, distribution, componentCategory, securityType, match[0].trim(), match[1].trim())
```

Comment:
Did not get this part. How are we splitting based on #?

Reply:
Ya this for splitting `test_class#test_name`, I have added a small test for both OpenSearch and Dashboards.